nengo_gui/components/pointer.py

```diff
@@ -70,8 +70,18 @@ def javascript(self):
         return 'new Nengo.Pointer(main, sim, %s);' % json
 
     def message(self, msg):
-        if len(msg) == 0:
+        if msg == ':empty:':
             self.override_target = None
+        elif msg[0:12] == ':check only:':
+            if len(msg) == 12:
+                self.data.append("good_pointer")
+            else:                
+                vocab = copy.deepcopy(self.vocab_out)
+                try:
+                    vocab.parse(msg[12:])
+                    self.data.append("good_pointer")
+                except:
```

Comment:
Right, I forgot that aspect of the vocab parsing. I'm okay with this now.

On Wed, Aug 5, 2015 at 9:01 PM, tcstewar <notifications@github.com> wrote:

> In nengo_gui/components/pointer.py
> <https://github.com/nengo/nengo_gui/pull/475#discussion_r36372803>:
>
> >              self.override_target = None
> > +        elif msg[0:12] == ':check only:':
> > +            if len(msg) == 12:
> > +                self.data.append("good_pointer")
> > +            else:
> > +                vocab = copy.deepcopy(self.vocab_out)
> > +                try:
> > +                    vocab.parse(msg[12:])
> > +                    self.data.append("good_pointer")
> > +                except:
>
> Hmm, I don't think there'll be a custom Nengo error for this. The problem
> is that there are a large number of different things that could be thrown
> there, depending on what bizarre thing someone gives to parse. And no
> matter what Exception someone manages to generate in there, we want to
> handle all of them with the response bad_pointer. So I think this is a
> legit use of a bare except.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nengo/nengo_gui/pull/475/files#r36372803>.
>